File: Leetcode/Easy/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        # 문자열 소문자 변환
        str = s.lower()

        # 영문, 숫자 외 문자 제거
        str = re.sub('[^a-z0-9]', '', str)

        # 팰린드롬 유무 판별 (발견하지 못했다면 True 반환 후 종료)
        return str == str[::-1]

File: Leetcode/Easy/valid_palindrome.java
class Solution {
    
    public boolean isPalindrome(String s) {
        
        int first = 0;
        int second = s.length()-1;
        
        while(first<second){
            while(first<second && !(Character.isLetterOrDigit(s.charAt(first)))) first++;
            while(first<second && !(Character.isLetterOrDigit(s.charAt(second)))) second--;
            
            if(Character.toLowerCase(s.charAt(first))!=Character.toLowerCase(s.charAt(second))) return false;
            
            first++;
            second--;
            
        }
        
        return true;
        
    }
}
